src/OpenTelemetry/Trace/ActivitySamplingParameters.cs

```diff
@@ -0,0 +1,85 @@
+// <copyright file="ActivitySamplingParameters.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// Sampling parameters passed to an <see cref="ActivitySampler"/> for it to make a sampling decision.
+    /// </summary>
+    public readonly struct ActivitySamplingParameters
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ActivitySamplingParameters"/> struct.
+        /// </summary>
+        /// <param name="parentContext">Parent activity context. Typically taken from the wire.</param>
+        /// <param name="traceId">Trace ID of a activity to be created.</param>
+        /// <param name="name">The name (DisplayName) of the activity to be created. Note, that the name of the activity is settable.
+        /// So this name can be changed later and Sampler implementation should assume that.
+        /// Typical example of a name change is when <see cref="Activity"/> representing incoming http request
+        /// has a name of url path and then being updated with route name when routing complete.
+        /// </param>
+        /// <param name="kind">The kind of the Activity to be created.</param>
+        /// <param name="tags">Initial set of Tags for the Activity being constructed.</param>
+        /// <param name="links">Links associated with the activity.</param>
+        public ActivitySamplingParameters(
```

Comment:
when we fit sampling into `GetRequestedDataUsingParentId` (we current do only `GetRequestedDataUsingContext`), the sampling parameter will not have ActivityContext, but only the parentid.
(just posting a note for later revisit)

Reply:
Except by the `TraceId` this becomes `ActivityCreationOptions<T>` depending on how the other issues are handled we may be able to use it directly.